tests/AutoReview/ProjectCodeTest.php

```diff
@@ -522,6 +522,62 @@ public function testAllCodeContainSingleClassy($className)
         }
     }
 
+    /**
+     * @dataProvider provideSrcClassCases
+     *
+     * @param string $className
+     */
+    public function testInheritdocIsNotAbused($className)
+    {
+        $rc = new \ReflectionClass($className);
+
+        $allowedMethods = array_map(
+            function (\ReflectionClass $interface) {
+                return $this->getPublicMethodNames($interface);
+            },
+            $rc->getInterfaces()
+        );
+
+        if (\count($allowedMethods)) {
+            $allowedMethods = array_merge(...array_values($allowedMethods));
+        }
+
+        $parentClass = $rc;
+        while (false !== $parentClass = $parentClass->getParentClass()) {
```

Comment:
should we also check interfaces and traits?

Reply:
We check interfaces, see line 538, about traits I'm not sure, can we use `@inheritdoc` to inherit from trait? I've never seen such situation.